src/mugs.js

```diff
@@ -1177,13 +1208,22 @@ define([
             } else {
                 constraintMsg = mug.p.constraintMsgAttr;
             }
+
+            var required;
+            if (mug.p.requiredAttr) {
+                required = mug.p.requiredCondition || util.createXPathBoolFromJS(mug.p.requiredAttr);
```

Comment:
It's not clear why `required` and `requiredCondition` attributes need to be saved separately.

I think `required` should be one of
- `"true()"` – _Required_ is checked, _Required Condition_ is empty
- `null` or `"false()"` – _Required_ is not checked
- XPath expression in _Required Condition_ – _Required_ is checked, _Required Condition_ is not empty

The value of `requiredCondition` can be inferred from the value of the `required` attribute on parsing:

- `"true()"` – _Required_ is checked, _Required Condition_ is empty
- not set or `"false()"` – _Required_ is not checked, _Required Condition_ is empty
- other XPath expression – _Required_ is checked, _Required Condition_ has value stored in `required` attribute.

Reply:
I did this because of the above conversation about being able to save the state of the condition without having the condition be active. Sorry for not making that clear. It seems like a nice thing to be able to disable that to test things without losing the value of that field

Your description here is what it was before this set of commits. Shall I revert 070c641 to make it that again?

Reply:
Ugh, sorry @jmtroth0, my preference is to use the algorithm outlined above. I think it's preferable to have a normalized and unambiguous source of truth in the serialized state rather than allowing the attributes to get out of sync and have ambiguous outcome. If we want to save the `requiredCondition` we could put its value in a `vellum:requiredCondition` attribute, which would be used as a hint for vellum only (`vellum:` attributes get stripped before XML is sent to the phone).

Reply:
I believe that is what this is doing, but I can check. Do you still have a preference against having this at all?

Reply:
Preserving it probably creates a better UX, so I guess it's a good idea to do that. And forget what I said about creating a `vellum:` attribute, that won't work in this context.